packages/odata-vocabularies/types/csdl.d.ts

```diff
@@ -0,0 +1,625 @@
+         *
+         * This interface was referenced by `NavigationPropertyBinding`'s JSON-Schema definition
+         * via the `patternProperty` ".*".
+         */
+        [k: string]: string;
+    }
+
+    /**
+     * Singleton
+     */
+    export interface Singleton extends UnboxContent<CSDLAnnotations> {
+        /**
+         * Qualified name of entity type
+         */
+        $Type: string;
+        /**
+         * Nullable singleton
+         */
+        $Nullable?: boolean;
+        $NavigationPropertyBinding?: NavigationPropertyBinding;
+    }
+
+    /**
+     * Action Import
+     */
+    export interface ActionImport extends UnboxContent<CSDLAnnotations> {
+        /**
+         * Qualified name of action
+         */
+        $Action: string;
+        /**
+         * Entity set of result entities
+         */
+        $EntitySet?: string;
+    }
+
+    /**
+     * Function Import
+     */
+    export interface FunctionImport extends UnboxContent<CSDLAnnotations> {
+        /**
+         * Qualified name of function
+         */
+        $Function: string;
+        /**
+         * Entity set of result entities
+         */
+        $EntitySet?: string;
+        /**
+         * Advertise in service document
+         */
+        $IncludeInServiceDocument?: boolean;
+    }
+
+    /**
+     * Action
+     */
+    export interface Action extends UnboxContent<CSDLAnnotations> {
+        /**
+         * Kind of model element
+         */
+        $Kind: 'Action';
+        /**
+         * Invoke on existing resource
+         */
+        $IsBound?: boolean;
+        /**
+         * Path to entity set of result entities
+         */
+        $EntitySetPath?: string;
+        $Parameter?: Parameter;
+        $ReturnType?: ReturnType;
+    }
+
+    /**
+     * Return Type
+     */
+    export interface ReturnType extends UnboxContent<CSDLAnnotations> {
+        /**
+         * Qualified name of return type
+         */
+        $Type?: string;
+        /**
+         * Returns a collection
+         */
+        $Collection?: boolean;
+        /**
+         * Can return null
+         */
+        $Nullable?: boolean;
+        $MaxLength?: MaxLength;
+        $Unicode?: Unicode;
+        $Precision?: Precision;
+        $Scale?: Scale;
+        $SRID?: SRID;
+    }
+
+    /**
+     * Function
+     */
+    export interface Function extends UnboxContent<CSDLAnnotations> {
+        /**
+         * Kind of model element
+         */
+        $Kind: 'Function';
+        /**
+         * Invoke on existing resource
+         */
+        $IsBound?: boolean;
+        /**
+         * Allows additional path segments or system query options
+         */
+        $IsComposable?: boolean;
+        /**
+         * Path to entity set of result entities
+         */
+        $EntitySetPath?: string;
+        $Parameter?: Parameter;
+        $ReturnType: ReturnType;
+    }
+}
+
+declare module '@sap-ux/vocabularies/CSDL' {
+    export = CSDL;
+}
```

Comment:
Why do we define this as a module with a name that looks like a package from us?

Reply:
This is a copy of https://github.com/SAP/open-ux-odata/blob/main/packages/vocabularies-types/utils/csdl.d.ts with some minor modifications. In the future we'd like to check if this can somehow be reused between the packages.

Reply:
Ok, thanks for clarification